tasks/mapt-oci/kind-aws-spot/deprovision/0.1/kind-aws-deprovision.yaml

```diff
@@ -0,0 +1,175 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: kind-aws-deprovision
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.44.x"
+    tekton.dev/categories: infrastructure
+    tekton.dev/tags: infrastructure, aws, kind
+    tekton.dev/displayName: "Kind Cloud Single Node - Destroy"
+    tekton.dev/platforms: "linux/amd64, linux/arm64"
+    mapt/version: "v0.9.1"
+spec:
+  description: |
+    Tears down a Kind cluster that was previously provisioned on AWS using the Mapt CLI.
+
+    This task is designed to safely and efficiently clean up cloud infrastructure created for ephemeral Kubernetes environments,
+    typically used in CI/CD pipelines or short-lived testing environments.
+
+    It ensures the following:
+    - Proper AWS credentials are used to authenticate the delete operation
+    - Cleanup is scoped to a specific environment ID to avoid accidental deletions
+    - Works seamlessly with previous `kind-aws-provision` task executions
+    - Supports a `debug` mode to enable verbose output for troubleshooting. !!WARNING!! Using debug mode can expose credentials.
+
+    Use this task in any workflow where you want to ensure that your test environments don't linger and incur unnecessary cloud costs.
+
+  volumes:
+    - name: aws-credentials
+      secret:
+        secretName: $(params.secret-aws-credentials)
+    - name: workdir
+      emptyDir: {}
+    - name: konflux-test-infra-volume
+      secret:
+        secretName: konflux-test-infra
+
+  stepTemplate:
+    env:
+      - name: KUBECONFIG
+        value: /var/workdir/.kube/config
+    volumeMounts:
+      - mountPath: /var/workdir
+        name: workdir
+
+  params:
+    - name: secret-aws-credentials
+      description: |
+        K8S secret holding the aws credentials. Secret should be accessible to this task.
+
+        ---
+        apiVersion: v1
+        kind: Secret
+        metadata:
+          name: aws-${name}
+        type: Opaque
+        data:
+          access-key: ${access_key}
+          secret-key: ${secret_key}
+          region: ${region}
+          bucket: ${bucket}
+    - name: id
+      description: |
+        A unique identifier for the Kind environment you want to destroy. This must match the ID used during creation.
+    - name: debug
+      description: |
+        If set to `true`, enables verbose output and displays command execution details.
+        Use only in secure environments as it may expose sensitive credentials.
+      default: 'false'
+    - name: pipeline-aggregate-status
+      type: string
+      description: |
+        The status of the pipeline (e.g., Succeeded, Failed, Completed, None).
+      default: None
+    - name: cluster-access-secret
```

Comment:
I would set a default value "" to this... as if no name is given for the secret the provision will create one with a random name....if here you do not set it (..is equal to "") it willl not be destroyed actively but when the pipelinerun is deleted...

I am saying this cause if you do not make this value optional...you are forcing to give a name on the provision and same on the decomission...and those could collide with names from other pipelines...

Setting is as optional...you still can set the name on provision and destroy but not forcing to that behavior

Reply:
otherwise LGTM

Reply:
cluster-access-secret is a param that will be mandatory to receive to collect logs etc from the cluster before shutting down

Reply:
right you needed for your use case